### PR TITLE
attempt at making a weather skills example

### PR DESCRIPTION
**Pull request by topherhansen**
### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [ ] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: samples/dotnet/kernel-syntax-examples/Skills/TravelPreparation/skprompt.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+In order to travel I need to know the weather in the city.
+I need to know what I should wear based on the weather.
+I need to know if I need additional items like an umbrella based on the weather.
+
+[EXAMPLES]
+If the weather is sunny, I should wear a t-shirt and sunglasses.
+If the weather is rainy, I should wear a raincoat and bring an umbrella.
+[END EXAMPLES]
+
+The weather is currently  {{$WEATHER_RESULT}} in my city. What should I wear and bring based on that?
```
Comment (anigo333) [inline samples/dotnet/kernel-syntax-examples/Skills/TravelPreparation/skprompt.txt:15]: Change this to:
What should I wear and bring

File: samples/dotnet/kernel-syntax-examples/Example43_TravelApp.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        string folder = @"C:\semantic-kernel\samples\dotnet\kernel-syntax-examples\Skills";//RepoFiles.SampleSkillsPath();
+        kernel.ImportSemanticSkillFromDirectory(folder,
+            "TravelPreparation");
+
+        using var weatherConnector = new WeatherApiConnector(Env.Var("WeatherApiKey"));
+        kernel.ImportSkill(new WeatherLookupSkill(weatherConnector), "weatherapi");
+
+        var planner = new SequentialPlanner(kernel);
+
+        var plan = await planner.CreatePlanAsync("Help me decide if I need an umbrella to walk to my neighbor's house. I live in Redmond Washington USA. Do I need an umbrella?");
```
Comment (anigo333) [inline samples/dotnet/kernel-syntax-examples/Example43_TravelApp.cs:43]: change this to:
I live in Seattle.

File: samples/dotnet/kernel-syntax-examples/Skills/TravelPreparation/skprompt.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+You are a travel planning agent. \
+You help people know what they should take or wear to prepare for the weather when walking between locations in the same city. \
+You are concise and only give the most important information. \
+You are not a weatherman.
+
+In order to travel I need to know the weather in the city.
+I need to know what I should wear based on the weather.
+I need to know if I need additional items like an umbrella based on the weather.
```
Comment (anigo333) [inline samples/dotnet/kernel-syntax-examples/Skills/TravelPreparation/skprompt.txt:8]: you are missing a function to answer that you should wear given the weather as an input

File: samples/dotnet/kernel-syntax-examples/Program.cs [hunk 4]
```diff
@@ -6,131 +6,11 @@
...
-
-        await Example17_ChatGPT.RunAsync();
-        Console.WriteLine("== DONE ==");
-
-        await Example18_DallE.RunAsync();
-        Console.WriteLine("== DONE ==");
 
-        await Example19_Qdrant.RunAsync();
+        await Example43_TravelApp.RunAsync(args[0]);
         Console.WriteLine("== DONE ==");
```
Comment (teresaqhoang) [inline samples/dotnet/kernel-syntax-examples/Program.cs:13]: Pls add other examples back

**Review by teresaqhoang — CHANGES_REQUESTED** [review]
This looks great! Left some comments. 

Tagging this with `PR: paused` since there hasn't been activity in 2 weeks. Please remove and address feedback if you want to continue with this work! If not, I'll close it out in a week if we don't get any response.

File: samples/dotnet/kernel-syntax-examples/Skills/Skills.Weather/WeatherLookupSkill.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+    public WeatherLookupSkill(IWeatherApiConnector connector)
+    {
+        this._connector = connector;
+    }
+
+    [SKFunction("Lookup the current weather for a location.")]
+    [SKFunctionName("LookupWeather")]
+    [SKFunctionInput(Description = "Location to get the weather for.")]
+    public async Task<string> SearchAsync(string location, SKContext context)
```
[the remark this answers is not in the document]
Comment (topherhansen) [reply samples/dotnet/kernel-syntax-examples/Skills/Skills.Weather/WeatherLookupSkill.cs:26]: Oh wow. I'm new to github and didn't realize I was creating a pull request to main. I hastily made a PR thinking it was going to my private branch while working on a hackathon.

I can clean up the code but not sure how quickly I'll be able to resolve all the comments.

[the remark this answers is not in the document]
Comment (teresaqhoang) [reply samples/dotnet/kernel-syntax-examples/Skills/Skills.Weather/WeatherLookupSkill.cs:26]: @topherhansen No worries! You can always close this PR out then create a new branch. All the changes should remain on your private fork!

**Comment by teresaqhoang** [comment]
I'm going to close this PR out since you didn't mean to open it up publicly against main. Feel free to re-open if you change your mind!

**Pull request closed without merging**